sources/packages/api/src/services/application/application.service.ts

```diff
@@ -675,35 +684,154 @@ export class ApplicationService extends RecordDataModelService<Application> {
-      .getMany();
+      });
+
+    // Available applications are records that have no 'scholastic standing reported' and are not 'archived'.
+    if (applicationStatus === ApplicationStatus.available) {
+      activeApplicationQuery.andWhere(
+        new Brackets((qb) => {
+          qb.where("studentScholasticStanding.id IS NULL").andWhere(
+            "application.isArchived = :isArchived",
+            {
+              isArchived: false,
+            },
+          );
+        }),
+      );
+    }
+
+    // Unavailable applications are records that are either 'completed (scholastic standing reported but not yet archived)' or 'archived'.
+    if (applicationStatus === ApplicationStatus.unavailable) {
+      activeApplicationQuery.andWhere(
+        new Brackets((qb) => {
+          qb.where("studentScholasticStanding.id IS NOT NULL").andWhere(
+            "application.isArchived = :isNotArchived",
+            {
+              isNotArchived: false,
+            },
+          );
+        }),
+      );
+      activeApplicationQuery.orWhere(
+        new Brackets((qb) => {
+          qb.where("application.isArchived = :isArchived", {
+            isArchived: true,
+          });
+        }),
+      );
+    }
+    activeApplicationQuery.orderBy("application.applicationNumber", "DESC");
+
+    if (paginationOptions.searchCriteria) {
+      activeApplicationQuery
+        .andWhere(
+          new Brackets((qb) => {
+            qb.where(
+              "CONCAT(user.firstName,' ', user.lastName) Ilike :searchCriteria",
+            ).orWhere("application.applicationNumber Ilike :searchCriteria");
+          }),
+        )
+        .setParameter(
+          "searchCriteria",
+          `%${paginationOptions.searchCriteria.trim()}%`,
+        );
+    }
+    activeApplicationQuery
+      .orderBy(
+        this.transformToEntitySortField(
+          paginationOptions.sortField,
+          paginationOptions.sortOrder,
+        ),
+      )
+      .offset(paginationOptions.page * paginationOptions.pageLimit)
+      .limit(paginationOptions.pageLimit);
+    const [result, count] = await activeApplicationQuery.getManyAndCount();
+    return {
+      results: result,
+      count: count,
+    };
+  }
+
+  /**
+   * Returns application status with respect to archive status and scholastic standing change.
+   * @param isArchived archive status.
+   * @param studentScholasticStandingId scholastic standing id.
+   * @returns application status.
+   */
+  getApplicationStatus(
+    isArchived: boolean,
+    studentScholasticStandingId: number,
+  ): ApplicationStatus {
+    if (isArchived) {
+      return ApplicationStatus.unavailable;
+    } else if (studentScholasticStandingId && !isArchived) {
+      return ApplicationStatus.completed;
+    } else if (!studentScholasticStandingId && !isArchived) {
+      return ApplicationStatus.available;
+    }
+  }
+
+  /**
+   **Transformation to convert the data table column name to database column name
```

Comment:
Please remove the double `**`. It does not make sense to have it applied in all the comment lines.

Reply:
@henryoforeh-aot  this change is still pending.